deepchem/models/layers.py

```diff
@@ -429,9 +427,9 @@ def _cosine_dist(x, y):
   y: tf.Tensor
     Input Tensor
   """
-  denom = (backend.sqrt(backend.sum(tf.square(x)) * backend.sum(tf.square(y))) +
-           backend.epsilon())
-  return backend.dot(x, tf.transpose(y)) / denom
+  x_norm = tf.nn.l2_normalize(x, axis=1)
+  y_norm = tf.nn.l2_normalize(y, axis=1)
+  return 1. - backend.dot(x_norm, tf.transpose(y_norm))
```

Comment:
Modified cosine distance function.

Reply:
I think this should be `backend.dot(x_norm, tf.transpose(y_norm))` instead of `1 - backend.dot(x_norm, tf.transpose(y_norm))` to compute the cosine distance right?

Reply:
Yea I think I was confused about whether we want the cosine similarity or the cosine distance... I think if the goal is to return the inner product of the normalized inputs then you're right, it shouldn't be `1 - backend.dot(x_norm, tf.transpose(y_norm))` but rather just `backend.dot(x_norm, tf.transpose(y_norm)`.